### PR TITLE
v6 updates

### PR DESCRIPTION
**Pull request by smaeda-ks**
https://twittercommunity.com/t/ads-api-version-6/129060

### CODE REVIEW

File: lib/twitter-ads/resources/analytics.rb [hunk 1]
```diff
@@ -173,11 +189,11 @@ def check_async_job_status(account, opts = {})
         # set default values
         job_ids = opts.fetch(:job_ids, nil)
         params = {}
-        params[:job_ids] = Array.wrap(job_ids).join(',') if job_ids
```
Comment (smaeda-ks) [inline lib/twitter-ads/resources/analytics.rb]: _bug_: Array#wrap is only available for ActiveSupport in Rails. This was introduced in https://github.com/twitterdev/twitter-ruby-ads-sdk/commit/b1f2e9764d262447464f2341b2a2af483f7635a4#diff-697bfc02b2bb3faa338ddbb781e0fb93 (2017)

File: lib/twitter-ads/account.rb [hunk 2]
```diff
@@ -257,6 +256,8 @@ def tailored_audiences(id = nil, opts = {})
     #
     # @since 0.2.3
     def scoped_timeline(id, opts = {})
+      TwitterAds::Utils.deprecated(
```
Comment (tushdante) [inline lib/twitter-ads/account.rb]: I believe the new Tweets endpoint is ready. So we can either publish this as is (so its in parity with the Python SDK) and make the change in another PR or we can just use the new Tweets endpoint here directly and remove the scoped timeline entirely. I'm fine either way, just wanted to make sure you're aware.

File: lib/twitter-ads/resources/analytics.rb [hunk 3]
```diff
@@ -25,36 +45,32 @@ module Analytics
...
-      # @see https://dev.twitter.com/ads/analytics/metrics-and-segmentation
-      # @since 1.0.0
-      def stats(metric_groups, opts = {})
-        self.class.stats(account, [id], metric_groups, opts)
-      end
-
+    # Pulls a list of metrics for the current object instance.
+    #
+    # @example
+    #   metric_groups = [:promoted_tweet_timeline_clicks, :promoted_tweet_search_clicks]
```
Comment (tushdante) [inline lib/twitter-ads/resources/analytics.rb]: Can you update this example here as well?

File: lib/twitter-ads/resources/analytics.rb [hunk 4]
```diff
@@ -210,17 +226,17 @@ def fetch_async_job_data(data_url)
...
       def active_entities(account, start_time:, end_time:, **opts)
-        entity_type = name
+        entity            = opts.fetch(:entity, name)
         granularity       = opts.fetch(:granularity, nil)
         start_utc_offset  = opts[:start_utc_offset] || opts[:utc_offset]
         end_utc_offset    = opts[:end_utc_offset] || opts[:utc_offset]
 
-        if entity_type == 'OrganicTweet'
+        if entity == 'OrganicTweet'
           raise "'OrganicTweet' not support with 'active_entities'"
```
Comment (tushdante) [inline lib/twitter-ads/resources/analytics.rb]: Nit: Looks like a minor typo. Error message should be `"'OrganicTweet' is not supported with 'active_entities'"`

File: lib/twitter-ads/account.rb (hunk 2 above)
Comment (smaeda-ks) [reply lib/twitter-ads/account.rb]: Thanks! Yes, I'll be adding the new Tweets endpoint in this PR and let you know again. Though, I'll leave this as is as some of the users might still need some time to migrate.

Comment (tushdante) [reply lib/twitter-ads/account.rb]: Awesome! Sounds good! Feel free to ping me for a review once you've made the changes

Comment (smaeda-ks) [reply lib/twitter-ads/account.rb]: @tushdante Added a few commits for the new Tweets endpoint support, thanks

File: lib/twitter-ads/resources/analytics.rb [hunk 5]
```diff
@@ -209,18 +225,46 @@ def fetch_async_job_data(data_url)
...
+      #   active_entities = TwitterAds::LineItem.active_entities(
+      #     account,
+      #     line_item_ids: %w(exrfs),
+      #     start_time: start_time,
+      #     end_time:   end_time,
+      #     utc_offset: utc_offset,
+      #     granularity: :day)
+      #
+      # @param account [Account] The Account object instance.
+      # @param entity [String] The entity type to retrieve data for.
```
Comment (tushdante) [inline lib/twitter-ads/resources/analytics.rb]: It sounds like with this implementation the `entity` is inferred by the calling class, correct? If so, should we update these docs to reflect the actual params available?

Comment (smaeda-ks) [reply lib/twitter-ads/resources/analytics.rb]: @tushdante Thanks for asking that, yes it was but I fixed it in this commit https://github.com/twitterdev/twitter-ruby-ads-sdk/pull/210/commits/6704ed5cf2ad05ae841d90e3ca7fdc157f97e327 so now it can be called directly with specifying `entity` parameter:

```ruby
TwitterAds::Analytics.active_entities( [..], entity: 'LINE_ITEM' )
```

Comment (tushdante) [reply lib/twitter-ads/resources/analytics.rb]: Oh gotcha, sounds good!
